uwblueprint/sistering
# Add VolunteerUser CRUD Methods

## Ticket link

Partially addresses #29 



## Implementation description
* Implement CRUD methods:
  - getVolunteerUserById
  - getVolunteerUserByEmail
  - getVolunteerUsers
  - createVolunteerUser
  - updateVolunteerUserById
  - deleteVolunteerUserById
  - deleteVolunteerUserByEmail

* Create corresponding GraphQL resolvers and DTOs



## Steps to test

1. Start docker and run `docker-compose up --build`
2. We need row(s) in the skills and branches tables for testing and will need to create those with raw SQL for now. Run the following commands:
`docker exec -it sistering_db /bin/bash`
`psql -U postgres -d sistering`
`INSERT INTO branches (name)
VALUES ('Branch1');`
`INSERT INTO skills (name)
VALUES ("Skill1");`

3. Run `docker exec -it sistering-backend /bin/bash -c "npx prisma generate"` and run the lastest migration files using `npx prisma migrate deploy`
4. Go to http://localhost:5000/graphql
5. We also need to register a user in order to hit the endpoints, so use the existing `register` endpoint first.
6. Then verify that all GraphQL queries and mutations introduced in this ticket work as expected



## What should reviewers focus on?
* Verify that GraphQL endpoints work as expected


## Checklist
